src/rest/handler.cpp

```diff
@@ -0,0 +1,473 @@
+        }
+        break;
+        case OT_NETWORK_DIAGNOSTIC_TLV_NETWORK_DATA:
+        {
+            value = FormatBytes(diagTlv.mData.mNetworkData.m8, diagTlv.mData.mNetworkData.mCount);
+            keyContainer.push_back("Network Data");
+            valueContainer.push_back(value);
+        }
+
+        break;
+        case OT_NETWORK_DIAGNOSTIC_TLV_IP6_ADDR_LIST:
+        {
+            std::vector<std::string> Ip6AddressList;
+            for (uint16_t i = 0; i < diagTlv.mData.mIp6AddrList.mCount; ++i)
+                Ip6AddressList.push_back(mJsonFormater.CreateIp6Address(diagTlv.mData.mIp6AddrList.mList[i]));
+            value = mJsonFormater.VectorToJson(Ip6AddressList);
+            keyContainer.push_back("IP6 Address List");
+            valueContainer.push_back(value);
+        }
+        break;
+        case OT_NETWORK_DIAGNOSTIC_TLV_MAC_COUNTERS:
+        {
+            value = mJsonFormater.CreateMacCounters(diagTlv.mData.mMacCounters);
+            keyContainer.push_back("MAC Counters");
+            valueContainer.push_back(value);
+        }
+        break;
+        case OT_NETWORK_DIAGNOSTIC_TLV_BATTERY_LEVEL:
+        {
+            std::unique_ptr<char> batteryLevel(new char(9));
+            sprintf(batteryLevel.get(), "%d", diagTlv.mData.mBatteryLevel);
+            value = batteryLevel.get();
+            keyContainer.push_back("Battery Level");
+            valueContainer.push_back(value);
+        }
+        break;
+        case OT_NETWORK_DIAGNOSTIC_TLV_SUPPLY_VOLTAGE:
+        {
+            std::unique_ptr<char> supplyVoltage(new char(9));
+            sprintf(supplyVoltage.get(), "%d", diagTlv.mData.mSupplyVoltage);
+            value = supplyVoltage.get();
+            keyContainer.push_back("Supply Voltage");
+            valueContainer.push_back(value);
+        }
+
+        break;
+        case OT_NETWORK_DIAGNOSTIC_TLV_CHILD_TABLE:
+        {
+            std::vector<std::string> ChildTableList;
+
+            for (uint16_t i = 0; i < diagTlv.mData.mChildTable.mCount; ++i)
+                ChildTableList.push_back(mJsonFormater.CreateChildTableEntry(diagTlv.mData.mChildTable.mTable[i]));
+            value = mJsonFormater.VectorToJson(ChildTableList);
+            keyContainer.push_back("Child Table");
+            valueContainer.push_back(value);
+        }
+        break;
+        case OT_NETWORK_DIAGNOSTIC_TLV_CHANNEL_PAGES:
+        {
+            value = FormatBytes(diagTlv.mData.mChannelPages.m8, diagTlv.mData.mChannelPages.mCount);
+            keyContainer.push_back("Channel Pages");
+            valueContainer.push_back(value);
+        }
+        break;
+        case OT_NETWORK_DIAGNOSTIC_TLV_MAX_CHILD_TIMEOUT:
+        {
+            std::unique_ptr<char> childTimeout(new char(9));
+
+            sprintf(childTimeout.get(), "%d", diagTlv.mData.mMaxChildTimeout);
+            value = childTimeout.get();
+            keyContainer.push_back("Max Child Timeout");
```

Comment:
It is not a good practice to include spaces in JSON key. You should define a table which maps TLV type to TLV name.